### PR TITLE
gdal2xyz.py improvements and documentation

### PR DESCRIPTION
**Pull request by idanmiara**
## What does this PR do?

gdal2xyz.py improvements and documentation

* skip_ndv - allow to skip lines with uninteresting data values
	True/False - skip lines with NDV data points;
	Sequence/Number - skip lines with given value (per band or per ds)
* support more flexible input options for srcfile, skip, band_nums
* type hinting added
* add --allBands command line option
* add main function

## Tasklist

 - [ ] Add test case(s)
 - [x] Add documentation
 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

### CODE REVIEW

File: gdal/swig/python/osgeo/utils/gdal2xyz.py [hunk 1]
```diff
@@ -78,9 +85,15 @@ def main(argv):
             band_nums.append(int(argv[i + 1]))
             i = i + 1
 
+        elif arg == '--allBands':
+            band_nums = None
```
Comment (rouault) [inline gdal/swig/python/osgeo/utils/gdal2xyz.py:89]: I find it a bit confusing that the semantics of [] or None is different. a allBands = True boolean would be more readable

Comment (idanmiara) [reply gdal/swig/python/osgeo/utils/gdal2xyz.py:89]: are you referring only to main() or also to gdal2xyz()?
I using another parameter for bands in gdal2xyz would be confusing.
Now that we use Python3, Another conversion that we could adopt is using Ellipsis (...) for "auto" or "default" value, i.e.        ```elif arg == '--allBands':
            band_nums = ...```

Comment (rouault) [reply gdal/swig/python/osgeo/utils/gdal2xyz.py:89]: let's forget my comment. I missed that the code was split in 2 functions. The code is OK as it

Comment (idanmiara) [reply gdal/swig/python/osgeo/utils/gdal2xyz.py:89]: thanks!

Comment (idanmiara) [reply gdal/swig/python/osgeo/utils/gdal2xyz.py]: @rouault I've addressed that issue you raised nonetheless.

**Comment by idanmiara** [comment]
Thanks!

**Pull request merged**